### PR TITLE
Join set do not loose ids of first set

### PR DESCRIPTION
**Pull request by pconesa**
This PR has 3 small changes:
1.- The join sets was losing the ids as soon as there were conflicts. Now, if allowed, it will keep the id of the first set. This will allow the usage of other protocols that us the Id for matching items, at least for the first set that is being merged.
2.- Fixes a "bug"  in the _idCount of the sets
3.- Closes the bd after loading the project info in the project list window.

### CODE REVIEW

File: pyworkflow/em/protocol/protocol_sets.py [hunk 1]
```diff
@@ -171,13 +172,15 @@ def createOutputStep(self):
...
+                    if not cleanIds or setNum == 1:
                         newObj.setObjId(objId)
                 else:
                     newObj = obj
 
-                if cleanIds:
+                if (cleanIds and setNum > 1) or self.renumber.get():
                     newObj.cleanObjId()
+                print ("FINAL ID: %s" % newObj.getObjId())
+                print ("OBJ ID: %s" % objId)
```
Comment (dmaluenda) [inline pyworkflow/em/protocol/protocol_sets.py]: Please, remove the prints or put them under debug mode

File: pyworkflow/em/constants.py [hunk 2]
```diff
@@ -1,3 +1,4 @@
+# -*- coding: utf-8 -*-
```
Comment (dmaluenda) [inline pyworkflow/em/constants.py]: I think is not needed

Comment (pconesa) [reply pyworkflow/em/constants.py]: It is! there is a degree symbol in line 99

File: pyworkflow/em/protocol/protocol_sets.py (hunk 1 above)
Comment (pconesa) [reply pyworkflow/em/protocol/protocol_sets.py]: upss

File: pyworkflow/em/constants.py (hunk 2 above)
Comment (dmaluenda) [reply pyworkflow/em/constants.py]: Ok! You right. However, as it's in a comment, I would change
```
# in the YZ plane such that rotation about the X axis by ~110° 
```
for 
```
# in the YZ plane such that rotation about the X axis by ~110deg.
```

**Comment by dmaluenda** [comment]
On the other hand, could I use this PR to introduce a bug fix when trying to join two SetOfCoordinates? The problem is _createSetOfCoordinates has a differnet signature than the _createSetOfAnything and it's fixed using the same that we did for the protocol_SubSet:
```
    # Overwrite SetOfCoordinates creation
    def _createSetOfCoordinates(self, suffix=''):
        coordSet = self.inputSets[0].get()
        micSet = coordSet.getMicrographs()
        return ProtSets._createSetOfCoordinates(self, micSet, suffix)
```

**Pull request merged**